Add 1.14 Forge basics (#2)
Mod-DevCafeTeam/MDC-Modding-Wiki: pages/1_14/getting_started/code_structure.md

of your mod's code.

## Reference

Let's start with the simplest, shortest but still important class.
This classes sole point is to store static, hardcoded values.
Most of the time all you will be storing here is your `MODID`

```java
class Reference {

    public static final String MODID = "examplemod";

}
```

## Main Class

This is probably the most important class in your mod.
While it often doesn't contain much code it is the main
entry point of your mod.

```java
@Mod(Reference.MODID)
public class ExampleMod {

    public static final Logger LOGGER = LogManager.getLogger();

    public ExampleMod() {
        IEventBus bus = FMLJavaModLoadingContext.get().getModEventBus();

        bus.addListener(this::setup);
        bus.addListener(this::clientSetup);
    }

    private void setup(final FMLCommonSetupEvent e) {
    }

    private void clientSetup(final FMLClientSetupEvent e) {
    }

}
```

As you can see for the basic set up you don't need much.

### @Mod
Let's start with the `@Mod` annotation. All you need
in here is your modid. As you can see we reference our `Reference`
class here. The modid can't be stored in the main class because

> then you wouldn't be able to use it in this annotation.

Reviewer: > The modid can't be stored in the main class because then you wouldn't be able to use it in this annotation.

Yes you can, it just needs to be const (aka `static final`) right ?

Reviewer 2: Oh ok looks like you can with a static import

It lets forge know that this class is the entry point of your mod.

### LOGGER

> The logger is used to log information useful for debbuging, resolving conflicts with other mods and notifying yourself if a user

Reviewer: ```suggestion
The logger is used to log information useful for debugging, resolving conflicts with other mods or just additional info for when a user
```

reports an issue.

### Constructor
This is what forge uses to get an instance of your main class.
In here you can register listeners for various lifecycle events
of your mod like `FMLCommonSetupEvent`, `FMLClientSetupEvent`
and so on.

### FMLCommonSetupEvent
This event (and thus the `#setup` function) can be used to perform any

> setup required on both sides. Most of the time it will be empty

Reviewer: If it's empty you don't need to have it at all though

Reviewer 2: I guess, but i usually leave it there just in case

entry point of your mod.

```java
package com.example.examplemod;

import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static com.example.examplemod.ExampleMod.MODID;

@Mod(MODID)
public class ExampleMod {

>     public static final Logger LOGGER = LogManager.getLogger();

Reviewer 3: Should the logger be gotten from the preinit event?

>     public static final Logger LOGGER = LogManager.getLogger();

Reviewer 3: Shouldn't this be gotten from the preinit event?

Reviewer 3: Oh actually forgot this is 1.14 heh, lemme know if this has actually changed and this is the way to do it.

Reviewer 2: the 1.14 example mod grabs the logger like this so i guess thats how you do it now
